app/controllers/profile/orders_controller.rb

```diff
@@ -18,4 +18,15 @@ def create
     @cart.empty
     redirect_to profile_path
   end
+
+  def update
+    order = Order.find(params[:id])
+      order.order_items.each do |item|
+        item.cancel
+      end
+    order.update(status: "cancelled")
```

Comment:
So my original thought was that this could be a Order method that does both the cancelling of the order items and the updating of the status of the order. That would really clean up the controller. This can be tabled as a later refactor though!